sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/config/CosmosConfig.java

```diff
@@ -33,7 +35,21 @@ public CosmosConfig(ResponseDiagnosticsProcessor responseDiagnosticsProcessor,
      * Initialization
      *
      * @param responseDiagnosticsProcessor must not be {@literal null}
-     * @param databaseThroughputConfig may be @{literal null}
+     * @param queryMetricsEnabled must not be {@literal null}
+     * @param maxDegreeOfParallelism must not be {@literal null}
+     */
+    @ConstructorProperties({"responseDiagnosticsProcessor", "queryMetricsEnabled", "maxDegreeOfParallelism"})
+    public CosmosConfig(ResponseDiagnosticsProcessor responseDiagnosticsProcessor,
```

Comment:
Do we need this new constructor? I think the one you added below is getting used in `ConfigBuilder`. 

We should try to expose as minimum public surface API as we can.

Reply:
You are correct, no usages found. Removed.